casper/src/main/scala/coop/rchain/casper/util/BondingUtil.scala

```diff
@@ -0,0 +1,164 @@
+    } yield s"""@"$ethAddress"!(["$pubKey", "$statusOut"], "${Base16.encode(unlockSig)}")"""
+  }
+
+  def walletTransferDeploy[F[_]: Sync](
+      nonce: Int,
+      amount: Long,
+      destination: String,
+      transferStatusOut: String,
+      pubKey: String,
+      secKey: Array[Byte]
+  )(implicit runtimeManager: RuntimeManager, scheduler: Scheduler): F[String] = {
+    val transferSigDataTerm =
+      mkTerm(s""" @"__SCALA__"!([$nonce, $amount, "$destination"].toByteArray())""").right.get
+
+    for {
+      sigBytes <- Sync[F].delay {
+                   runtimeManager
+                     .captureResults(runtimeManager.emptyStateHash, transferSigDataTerm)
+                     .head
+                     .exprs
+                     .head
+                     .getGByteArray
+                     .toByteArray
+                 }
+      transferSigData = Blake2b256.hash(sigBytes)
+      transferSig     = Secp256k1.sign(transferSigData, secKey)
+    } yield s"""
+             |for(@wallet <- @"$pubKey") {
+             |  @(wallet, "transfer")!($amount, $nonce, "${Base16.encode(transferSig)}", "$destination", "$transferStatusOut")
+             |}""".stripMargin
+  }
+
+  def writeFile[F[_]: Sync](name: String, content: String): F[Unit] = {
+    val file =
+      Resource.make[F, PrintWriter](Sync[F].delay { new PrintWriter(name) })(
+        pw => Sync[F].delay { pw.close() }
+      )
+    file.use(pw => Sync[F].delay { pw.println(content) })
+  }
+
+  def makeRuntimeDir[F[_]: Sync]: Resource[F, Path] =
+    Resource.make[F, Path](Sync[F].delay { Files.createTempDirectory("casper-bonding-helper-") })(
+      runtimeDir => Sync[F].delay { runtimeDir.recursivelyDelete() }
+    )
+
+  def makeRuntimeResource[F[_]: Sync](runtimeDirResource: Resource[F, Path]): Resource[F, Runtime] =
+    runtimeDirResource.flatMap(
+      runtimeDir =>
+        Resource
+          .make(Sync[F].delay { Runtime.create(runtimeDir, 1024L * 1024 * 1024) })(
+            runtime => Sync[F].delay { runtime.close() }
+          )
+    )
+
+  def makeRuntimeManagerResource[F[_]: Sync](
+      runtimeResource: Resource[F, Runtime]
+  ): Resource[F, RuntimeManager] =
+    runtimeResource.flatMap(
+      activeRuntime =>
+        Resource.make(RuntimeManager.fromRuntime(activeRuntime).pure[F])(_ => Sync[F].unit)
+    )
+
+  def writeRhoFiles[F[_]: Sync](
+      bondKey: String,
+      ethAddress: String,
+      amount: Long,
+      secKey: String,
+      pubKey: String
+  )(implicit scheduler: Scheduler): F[Unit] = {
+    val runtimeDirResource     = makeRuntimeDir[F]
+    val runtimeResource        = makeRuntimeResource[F](runtimeDirResource)
+    val runtimeManagerResource = makeRuntimeManagerResource[F](runtimeResource)
+    runtimeManagerResource.use(
+      runtimeManager =>
+        for {
+          unlockCode <- unlockDeploy[F](ethAddress, pubKey, secKey)(
+                         Sync[F],
```

Comment:
I thought Sync gets passed in implicitly given how you already are writing `unlockDeploy[F]`